### PR TITLE
feat: add `merge_build_and_host_envs` section

### LINKED ISSUE #518: Implement `merge_build_and_host_envs`
Was: `merge_build_host` in `conda-build`.

This option would just combine the specs from build and host env and create a joint "host" environment.

This is used in many `R` recipes on Windows: https://github.com/search?q=org%3Aconda-forge+merge_build_host%3A+true+language%3AYAML&type=code&l=YAML

### PR DESCRIPTION
**Pull request by swarnimarun**
- Added the `merge_build_and_host_envs` section as named in `schema.json`.
- It merges build into host env. But leaves the build env untouched. 
- Run dependencies are built from new host env.
- Modified the test to use `merge_build_and_host_envs`

Fixes #518

### CODE REVIEW

File: src/recipe/parser/build.rs [hunk 1]
```diff
@@ -55,9 +55,17 @@ pub struct Build {
     /// Setting to control wether to always include a file (even if it is already present in the host env)
     #[serde(default, skip_serializing_if = "GlobVec::is_empty")]
     pub(super) always_include_files: GlobVec,
+    /// Merge the build and host envs
+    #[serde(default)]
```
Comment (wolfv) [inline src/recipe/parser/build.rs:59]: Would be cool to skip rendering this if it's false (the default).

I think I added a `is_false` function somewhere. We could move it to a `serde_utils.rs` file.

Comment (swarnimarun) [reply src/recipe/parser/build.rs]: We can use `std::ops::Not::not`

File: src/render/resolved_dependencies.rs [hunk 2]
```diff
@@ -594,6 +596,31 @@ pub async fn resolve_dependencies(
         None
     };
 
+    let host_env = if merge_build_host {
```
Comment (wolfv) [inline src/render/resolved_dependencies.rs:599]: I think this is the wrong place. The merging should happen of the `matchspecs`, and before the environment is resolved. 

E.g. if you have

```
build:
  - compiler
host:
  - bla
```

It should just insert `compiler` into the `host` env (before resolving & installing) and then resolve everything at once into the same environment. We shouldn't have to touch the `ResolvedDependencies` as they should be fine (at that stage).

```
    let match_specs = specs.iter().map(|s| s.spec().clone()).collect::<Vec<_>>();
    // insert build match specs here!
    let host_env = if !match_specs.is_empty() {
```

**Review by wolfv — CHANGES_REQUESTED** [review]
There is a little logic bug on where `host` and `build` deps are merged. It needs to happen at the `matchspec` level :)

File: src/render/resolved_dependencies.rs (hunk 2 above)
Comment (swarnimarun) [reply src/render/resolved_dependencies.rs:599]: oh, that makes sense. Yeah pretty easy to change.

File: src/render/resolved_dependencies.rs [hunk 3]
```diff
@@ -553,7 +553,16 @@ pub async fn resolve_dependencies(
         }
     }
 
-    let match_specs = specs.iter().map(|s| s.spec().clone()).collect::<Vec<_>>();
+    let match_specs = specs
```
Comment (wolfv) [inline src/render/resolved_dependencies.rs:556]: this looks fine but should only happen when the `merge_build_host` is true. 
Also in that case, the build env does not need to be created.

**Review by wolfv — CHANGES_REQUESTED** [review]
still some logic fixes necessary

**Comment by wolfv** [comment]
It also still creates the build env ... We want to merge build and host, and therefore don't need to create the build env at all.

File: src/render/resolved_dependencies.rs [hunk 4]
```diff
@@ -663,26 +674,30 @@ pub async fn resolve_dependencies(
...
-                    run_specs
-                        .depends
-                        .extend(clone_specs(name, "build", &rex.strong)?);
-                    run_specs.constrains.extend(clone_specs(
-                        name,
-                        "build",
-                        &rex.strong_constrains,
-                    )?);
+    // if already merged don't redo
+    if !merge_build_host {
```
Comment (wolfv) [inline src/render/resolved_dependencies.rs:678]: this isnt' reallt necessary anymore, right? Since the build env is None now.

**Review by wolfv — APPROVED** [review]

**Pull request merged**